mender-update/standalone.hpp

```diff
@@ -0,0 +1,125 @@
+// Copyright 2023 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+#ifndef MENDER_UPDATE_STANDALONE_HPP
+#define MENDER_UPDATE_STANDALONE_HPP
+
+#include <unordered_map>
+
+#include <common/error.hpp>
+#include <common/expected.hpp>
+#include <common/key_value_database.hpp>
+#include <common/json.hpp>
+#include <common/optional.hpp>
+
+#include <artifact/artifact.hpp>
+
+#include <mender-update/context.hpp>
+
+#include <mender-update/update_module/v3/update_module.hpp>
+
+namespace mender {
+namespace update {
+namespace standalone {
+
+using namespace std;
+
+namespace database = mender::common::key_value_database;
+namespace error = mender::common::error;
+namespace expected = mender::common::expected;
+namespace json = mender::common::json;
+namespace optional = mender::common::optional;
+
+namespace artifact = mender::artifact;
+
+namespace context = mender::update::context;
+
+namespace update_module = mender::update::update_module::v3;
+
+// The keys and data, respectively, of the JSON object living under the `standalone_data_key` entry
+// in the database. Be sure to take into account upgrades when changing this.
+struct DataKeys {
+	static const string version;
+	static const string artifact_name;
+	static const string artifact_group;
+	static const string artifact_provides;
+	static const string artifact_clears_provides;
+	static const string payload_types;
+};
```

Comment:
Maybe this could be just a singleton variable and not a type?

Reply:
Doesn't really make much difference though, does it?

Reply:
No, it doesn't. It just looks a bit weird to me to create an instance of a type only to have access to 4 constants.